nengo/builder/signal.py

```diff
@@ -299,25 +386,31 @@ def __str__(self):
         return sio.getvalue()
 
     def init(self, signal):
-        """Set up a permanent mapping from signal -> ndarray."""
+        """Set up a permanent mapping from signal -> data."""
         if signal in self:
             raise SignalError("Cannot add signal twice")
 
-        x = signal.initial_value
-        if signal.is_view:
-            if signal.base not in self:
-                self.init(signal.base)
-
-            # get a view onto the base data
-            view = np.ndarray(
-                shape=x.shape, strides=x.strides, offset=signal.offset,
-                dtype=x.dtype, buffer=self[signal.base].data)
-            assert np.array_equal(view, x)
-            view.setflags(write=not signal.readonly)
-            dict.__setitem__(self, signal, view)
-        else:
-            x = x.view() if signal.readonly else x.copy()
```

Comment:
Yes, I think this is true for now. It's actually not clear to me if Scipy sparse matrices can be views (it seems to me that it should be possible to share the same data and get a view, but I haven't figured out how to do it), so for now I've just disallowed it with `SparseSignal`. At some point we might want to move some of this logic for getting views into `Signal` itself, so that it can be done differently depending on the type of signal. But for now, I think it's fine to assume all signals that can have views are based on Numpy arrays.